src/transformation/visitors/typeof.ts

```diff
@@ -15,37 +15,37 @@ export function transformTypeOfBinaryExpression(
     node: ts.BinaryExpression
 ): lua.Expression | undefined {
     const operator = node.operatorToken.kind;
-    function transformTypeOfLiteralComparison(
-        typeOfExpression: ts.TypeOfExpression,
-        comparedExpression: lua.StringLiteral
-    ): lua.Expression {
-        if (comparedExpression.value === "object") {
-            comparedExpression.value = "table";
-        } else if (comparedExpression.value === "undefined") {
-            comparedExpression.value = "nil";
-        }
+    if (
+        operator !== ts.SyntaxKind.EqualsEqualsToken &&
+        operator !== ts.SyntaxKind.EqualsEqualsEqualsToken &&
+        operator !== ts.SyntaxKind.ExclamationEqualsToken &&
+        operator !== ts.SyntaxKind.ExclamationEqualsEqualsToken
+    ) {
+        return;
```

Comment:
I assume you're missing a '// TypeScript error' here?

Reply:
That's actually not an error, it's valid to do `typeof x > 'string'`, and to make it the same as JS it has to fall back to a helper call

Reply:
No, when `transformTypeOfBinaryExpression` returns `undefined` it just means that there is no comparison optimization, so it fallbacks to regular `typeof` expression transform (`transformTypeOfExpression`)